### PR TITLE
Open loop testing mode

### PR DESCRIPTION
**Pull request by oschaaf**
Adds `--open-loop`, which will make the benchmark client refrain from applying backpressure. Instead, stream creation is unconditional and will be handled by the connection pool.
In practice this means we'll not be measuring time spend blocking,
instead we'll see overflow counters increases.

Part of https://github.com/envoyproxy/nighthawk/issues/25
Prerequisite: https://github.com/envoyproxy/nighthawk/pull/178

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

### CODE REVIEW

File: source/client/benchmark_client_impl.cc [hunk 1]
```diff
@@ -69,7 +71,9 @@ StatisticPtrMap BenchmarkClientHttpImpl::statistics() const {
 
 bool BenchmarkClientHttpImpl::tryStartRequest(CompletionCallback caller_completion_callback) {
   const uint64_t max_in_flight =
-      max_pending_requests_ + (use_h2_ ? max_active_requests_ : connection_limit_);
+      provide_resource_backpressure_
+          ? max_pending_requests_ + (use_h2_ ? max_active_requests_ : connection_limit_)
+          : UINT64_MAX;
```
Comment (eric846) [inline source/client/benchmark_client_impl.cc]: What about:

```
if (provide_resource_backpressure_) {
  const uint64_t max_in_flight =
      max_pending_requests_ + (use_h2_ ? max_active_requests_ : connection_limit_);

  if (requests_initiated_ - requests_completed_ >= max_in_flight) {
    // When we allow client-side queueing, we want to have a sense of time spend waiting on that
    // queue. So we return false here to indicate we couldn't initiate a new request.
    return false;
  }
}
```

File: test/integration/test_integration_basics.py [hunk 2]
```diff
@@ -109,6 +114,30 @@ def test_http_h2_mini_stress_test_without_client_side_queueing(http_test_server_
   assertNotIn("upstream_rq_pending_overflow", counters)
 
 
+def test_http_h1_mini_stress_test_open_loop(http_test_server_fixture):
+  """
+  H1 open loop stress test. We expect higher pending and overflow counts 
+  """
+  counters = mini_stress_test(http_test_server_fixture, [
+      http_test_server_fixture.getTestServerRootUri(), "--rps", "2500", "--max-pending-requests",
+      "1", "--duration 10", "--open-loop", "--max-active-requests", "1", "--connections", "1"
```
Comment (eric846) [inline test/integration/test_integration_basics.py]: Nit: I think you split all the others yesterday to `"--duration", "10"`

File: source/client/benchmark_client_impl.cc (hunk 1 above)
Comment (oschaaf) [reply source/client/benchmark_client_impl.cc]: Much nicer indeed -- I'm sold :-) Done in 0fcba50e4bbd0a645251ec81a9e377562aa433e6

File: test/integration/test_integration_basics.py (hunk 2 above)
Comment (oschaaf) [reply test/integration/test_integration_basics.py]: Fixed in 0fcba50e4bbd0a645251ec81a9e377562aa433e6

**Comment by oschaaf** [comment]
comments addressed!
/assign @eric846

File: README.md [hunk 3]
```diff
@@ -66,6 +66,10 @@ format>
 
 Where:
 
+--open-loop
+Enable open loop mode. When enabled, the benchmark client will not
+provide backpressurewhen resource limits are hit.
```
[the remark this answers is not in the document]
Comment (oschaaf) [reply README.md]: Amended

**Pull request merged**